src/scm/gitlab.js

```diff
@@ -0,0 +1,278 @@
+                    resolve({file: tree.path, content: decodeURIComponent(escape(atob(content.content)))});
+                  })
+                  .fail(reject)
+              });
+            });
+          return Promise.all(promises);
+        });
+  }
+
+  getNamespaces() {
+    let testUrl = `${this.baseUrl}/groups?${this.tokenParam}`;
+    return getAllItems(Promise.resolve(
+      {
+        token: this.accessToken,
+        items: [],
+        url: testUrl
+      }),
+      this.followPaginate,
+      'gitlab'
+    )
+      .then(groups => {
+        this.namespaces = [this.user].concat(groups.map(group => group.name));
+        return this.namespaces;
+      })
+      .catch((err) => {
+        showAlert('Failed to get user info.', LEVEL_ERROR);
+      });
+  }
+
+  getRepos() { // Named Projects in gitlab
+    return getAllItems(Promise.resolve(
+      {
+        token: this.accessToken,
+        items: [],
+        url: `${this.baseUrl}/users/${this.user}/projects?${this.tokenParam}`
+      }),
+      this.followPaginate,
+      'gitlab'
+    )
+      .then(response => {
+        this.namesToIds.repos = response.reduce((obj, item) => (obj[item.name] = item.id, obj), {});
+        const repos = Object.keys(this.namesToIds.repos);
+        //if current bind still existed, use it
+        const repo = context.bindRepo[context.id];
+        if (repo && $.inArray(repo.fullName, repos) >= 0) {
+          context.repo = repo;
+        }
+        return repos;
+      });
+  }
+
+  createRepo() {
+    const owner = $('#new-repo-owner').val();
+    const name = $('#new-repo-name').val();
+    const desc = $('#new-repo-desc').val();
+    const visibility = ($('#new-repo-type').val() !== 'public') ? 'private' : 'public';
+    const payload = {
+      name: name,
+      description: desc,
+      visibility: visibility
+    };
+    if (!name || name === '') return;
+    return new Promise((resolve, reject) => {
+      return $.ajax({
+        url: `${this.baseUrl}/projects`,
+        headers: this.tokenHeader,
+        method: 'POST',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      })
+        .then(resolve)
+        .fail(reject);
+    })
+      .then(response => {
+        const repo = {
+          fullName: response.name,
+          id: response.id
+        };
+        context.repo = repo;
+        Object.assign(context.bindRepo, {[context.id]: repo});
+        if (context.bindBranch[context.id]) {
+          delete context.bindBranch[context.id];
+        }
+        chrome.storage.sync.set({bindRepo: context.bindRepo});
+        return response.name;
+      })
+      .then(repo => {
+        return this.commitFiles(repo, 'master', null, [{name: "README.md", content: "initialed by gas-github"}], null,null, 'initial commit')
+          .then(() => {
+            return repo;
+          });
+      })
+      .catch((err) => {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/5becb274ac17be98fee6e945e23cbab998e9d302/src/scm/gitlab.js#L226"></span>‘err’ is defined but never used.</td>
    </tr>
  </tbody>
</table>